x/axelarnet/rate_limit.go

```diff
@@ -0,0 +1,95 @@
+package axelarnet
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	ibctransfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	porttypes "github.com/cosmos/ibc-go/v3/modules/core/05-port/types"
+	ibcexported "github.com/cosmos/ibc-go/v3/modules/core/exported"
+
+	"github.com/axelarnetwork/axelar-core/x/axelarnet/keeper"
+	"github.com/axelarnetwork/axelar-core/x/axelarnet/types"
+	nexustypes "github.com/axelarnetwork/axelar-core/x/nexus/exported"
+	"github.com/axelarnetwork/utils/funcs"
+)
+
+// RateLimiter implements an ICS4Wrapper middleware to rate limit IBC transfers
+type RateLimiter struct {
+	keeper  keeper.Keeper
+	channel porttypes.ICS4Wrapper
+	nexus   types.Nexus
+}
+
+// NewRateLimiter returns a new RateLimiter
+func NewRateLimiter(keeper keeper.Keeper, channel porttypes.ICS4Wrapper, nexus types.Nexus) RateLimiter {
+	return RateLimiter{
+		keeper:  keeper,
+		channel: channel,
+		nexus:   nexus,
+	}
+}
+
+// RateLimitPacket implements rate limiting of IBC packets
+// - If the IBC channel that the packet is sent on is a registered chain, check the activation status.
+// - If the packet is an ICS-20 coin transfer, apply rate limiting on (chain, base denom) pair.
+// - If the rate limit is exceeded, an error is returned.
+func (r RateLimiter) RateLimitPacket(ctx sdk.Context, packet ibcexported.PacketI, direction nexustypes.TransferDirection) error {
+	ibcPath := fmt.Sprintf("%s/%s", packet.GetSourcePort(), packet.GetSourceChannel())
+	chainName, ok := r.keeper.GetChainNameByIBCPath(ctx, ibcPath)
+	if !ok {
+		// TODO: if IBC channel is not registered, use axelarnet for activation/deactivation state
```

Comment:
I don't understand what this todo means

Reply:
that was in regards to using axelarnet deactivation as full IBC deactivation. Removed the comment for now